### PR TITLE
Remove deprecated Cajita

### PR DESCRIPTION
**Pull request by streeve**
Replace deprecated wrappers and add variadic grid halos

### CODE REVIEW

**Comment by streeve** [comment]
Need to test on Summit before we merge

**Comment by streeve** [comment]
ping @kwitaechong. I forgot about this, but this should be updated

File: src/ExaMPM_TimeIntegrator.hpp [hunk 1]
```diff
@@ -252,8 +250,7 @@ void g2p( const ExecutionSpace& exec_space, const ProblemManagerType& pm,
     Kokkos::Experimental::contribute( k_c, k_c_sv );
 
     // Complete global scatter.
-    pm.scatter( Location::Cell(), Field::Density() );
-    pm.scatter( Location::Cell(), Field::Mark() );
+    pm.scatter( Location::Cell() );
```
Comment (kwitaechong) [inline src/ExaMPM_TimeIntegrator.hpp:253]: Likewise, at line 320, 323

pm.scatter( Location::Node() );
pm.gather( Location::Node() ); ??

Comment (streeve) [reply src/ExaMPM_TimeIntegrator.hpp:253]: It was Cell before though - maybe I'm misunderstanding

**Comment by streeve** [comment]
@kwitaechong can you re-review?

**Review by kwitaechong — APPROVED** [review]

**Pull request merged**